CMakeLists.txt

```diff
@@ -0,0 +1,713 @@
+check_function_exists(fflush   HAVE_FFLUSH_POINTER)
+check_function_exists(fnmatch  HAVE_FNMATCH)
+check_function_exists(fopen    HAVE_FOPEN_POINTER)
+check_function_exists(fpclass  HAVE_FPCLASS)
+check_function_exists(fprintf  HAVE_FPRINTF_POINTER)
+check_function_exists(fread    HAVE_FREAD_POINTER)
+check_function_exists(fseek    HAVE_FSEEK_POINTER)
+check_function_exists(ftell    HAVE_FTELL_POINTER)
+check_function_exists(fwrite   HAVE_FWRITE_POINTER)
+check_function_exists(setvbuf  HAVE_SETVBUF_POINTER)
+check_function_exists(strerror HAVE_STRERROR)
+
+##
+# check_symbol_exists
+##
+check_symbol_exists(isnan "math.h" HAVE_ISNAN)
+check_symbol_exists(memmove "memory.h" HAVE_MEMMOVE)
+check_symbol_exists(add_history "readline.h" HAVE_READLINE_HISTORY)
+
+
+
+###-----------------------------------------------------------------------------
+# Set flags & options from config-site files after everything is set up since
+# setting them beforehand can upset some of CMake's own find routines. 
+###-----------------------------------------------------------------------------
+if(DEFINED SILO_C_FLAGS)
+    string(APPEND CMAKE_C_FLAGS " ${SILO_C_FLAGS}")
+endif()
+
+if(DEFINED SILO_CXX_FLAGS)
+    string(APPEND CMAKE_CXX_FLAGS " ${SILO_CXX_FLAGS}")
+endif()
+
+# will this be needed?
+#if(DEFINED SILO_EXE_LINKER_FLAGS)
+#    string(APPEND CMAKE_EXE_LINKER_FLAGS " ${SILO_EXE_LINKER_FLAGS}")
+#endif()
+
+
+###-----------------------------------------------------------------------------
+# Create silo.h from silo.h.in
+###-----------------------------------------------------------------------------
+set(SILO_VERS_MAJ ${Silo_VERSION_MAJOR})
+set(SILO_VERS_MIN ${Silo_VERSION_MINOR})
+set(SILO_VERS_PAT ${Silo_VERSION_PATCH})
+set(SILO_VERS_TAG Silo_version_${SILO_VERS_MAJ}_${SILO_VERS_MIN})
+if(SILO_VERS_PAT)
+    string(APPEND SILO_VERS_TAG "_${SILO_VERS_PAT}")
+endif()
+
+configure_file(${Silo_SOURCE_DIR}/src/silo/silo.h.in
+               ${silo_build_include_dir}/silo.h)
+
+
+###-----------------------------------------------------------------------------
+# Create config.h from config.h.in
+###-----------------------------------------------------------------------------
+
+if(SILO_ENABLE_FORTRAN)
+    enable_language(Fortran)
+    include(FortranCInterface)
+    # for config.h
+    FortranCInterface_HEADER(${silo_build_include_dir}/silo_FC.h MACRO_NAMESPACE "FC_")
+    if(EXISTS ${silo_build_include_dir}/silo_FC.h)
+        set(HAVE_SILO_FC_H 1)
+        install(FILES ${silo_build_include_dir}/silo_FC.h DESTINATION include
+                PERMISSIONS OWNER_READ OWNER_WRITE
+                            GROUP_READ GROUP_WRITE
+                            WORLD_READ) 
+    endif()
+endif()
+
+# set up some vars to be passed to config.h.in
+set(SILO_VERSION ${Silo_VERSION})
+set(SILO_BUG "miller86@llnl.gov")
+set(SILO_NAME "silo")
+set(SILO_TARNAME "silo")
+set(SILO_URL "https://wci.llnl.gov/simulation/computer-codes/silo")
```

Comment:
Lets just shorten to `https://silo.llnl.gov`

Reply:
done